pkgs/tools/admin/azure-cli/python-packages.nix

```diff
@@ -30,6 +32,16 @@ let
 
         sourceRoot = "${src.name}/src/azure-cli-core";
 
+        patches =
+          # Adding the possibility to configure an immutable configuration dir via `AZURE_IMMUTABLE_DIR`.
+          # This enables us to place configuration files that alter the behavior of the CLI in the Nix store.
+          #
+          # This is a downstream patch without an commit or PR upstream.
+          # There is an issue to discuss possible solutions upstream:
+          # https://github.com/Azure/azure-cli/issues/28093
+          lib.optional withImmutableConfig ./0001-optional-immutable-configuration-dir.patch
```

Comment:
Can we make that depend purely on an env?

Reply:
Not sure I get your question. The patch already checks for the env. Are you suggesting to just always apply it?

Reply:
> Are you suggesting to just always apply it?

Yes, I think that would be a lot easier

Reply:
Done. :)